File: files/400-499/446/CH-446.R
library(tidyverse)
library(readxl)

path <- "400-499/446/CH-446 Number Series.xlsx"
input <- read_excel(path, range = "B3:C10")
test <- read_excel(path, range = "E3:F10")

result <- input %>%
  mutate(
    Sale = case_when(
      str_detect(Sale, "\\d") ~ as.numeric(Sale),
      TRUE ~ NA_integer_
    )
  ) %>%
  mutate(Sale = zoo::na.approx(Sale))

all.equal(result, test)
# True

File: files/400-499/446/CH-446.py
import pandas as pd

path = "400-499/446/CH-446 Number Series.xlsx"
input = pd.read_excel(path, usecols="B:C", nrows=7, skiprows=2)
test = pd.read_excel(path, usecols="E:F", nrows=7, skiprows=2)
test.columns = input.columns

input["Sale"] = (
    pd.to_numeric(input["Sale"], errors="coerce").interpolate().astype("int64")
)
print(input.equals(test))
# True
